### PR TITLE
feat(designer): Adding RunAfter button for right clicking on actions

### PR DESCRIPTION
**Pull request by spanvalkar**
- **Please check if the PR fulfills these requirements**

* [x] The commit message follows our guidelines
* [x] Tests for the changes have been added (for bug fixes/features)
* [x] Docs have been added / updated (for bug fixes / features)

- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Feature

- **What is the current behavior?** (You can also link to an open issue here)

[Feature 25928538](https://msazure.visualstudio.com/One/_workitems/edit/25928538): Right click 'Run after' on an action

Feature ADO link is above. 

We've heard feedback from V3 Designer users (in Power Automate) that they would like help understanding the dependencies that a given operation has. Thankfully, we already have this dependency data; it is in the 'Settings' tab on the node panel and is called 'Run After' data.

- **What is the new behavior (if this is a feature change)?**

This feature adds a new button when a user right-clicks on an action in a flow that opens the node panel to the 'settings' tab, revealing the 'Run After' data so that users can see the dependencies for a given action. 

Note: This button only appears if there is a 'RunAfter' dependency for a given action. See screenshots below for clarification.

- **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

There shouldn't be any breaking changes. This PR introduces new functionality to allow users to see the action that a given operation runs after, so it does not affect any existing functionality nor flows.

- **Please Include Screenshots or Videos of the intended change**:

RunAfter is visible for actions that have a 'RunAfter' dependency in the 'Settings' tab:

image

RunAfter is not visible for actions that do not have a 'RunAfter' dependency in the 'Settings' tab:

image

### CODE REVIEW

File: libs/designer/src/lib/ui/CustomNodes/OperationCardNode.tsx [hunk 1]
```diff
@@ -86,6 +88,7 @@ const DefaultNode = ({ targetPosition = Position.Top, sourcePosition = Position.
   const repetitionName = getRepetitionName(parentRunIndex, id, nodesMetaData, operationsInfo);
   const isSecureInputsOutputs = useSecureInputsOutputs(id);
   const { status: statusRun, error: errorRun, code: codeRun, repetitionCount } = runData ?? {};
+  const panelTabName = 'SETTINGS';
```
Comment (hartra344) [inline libs/designer/src/lib/ui/CustomNodes/OperationCardNode.tsx:91]:
```suggestion
  const RUN_AFTER_PANEL_TAB = 'SETTINGS';
```

File: libs/designer/src/lib/ui/CustomNodes/OperationCardNode.tsx [hunk 2]
```diff
@@ -213,6 +216,15 @@ const DefaultNode = ({ targetPosition = Position.Top, sourcePosition = Position.
     }
   }, [dispatch, id, nodeSelectCallbackOverride, suppressDefaultNodeSelect]);
 
+  const nodeClick = useCallback(() => {
+    handleNodeSelection();
+  }, [handleNodeSelection]);
+
+  const runAfterClick = useCallback(() => {
+    handleNodeSelection();
+    dispatch(selectPanelTab(panelTabName));
```
Comment (hartra344) [inline libs/designer/src/lib/ui/CustomNodes/OperationCardNode.tsx:225]:
```suggestion
    dispatch(selectPanelTab(RUN_AFTER_PANEL_TAB));
```

File: libs/designer/src/lib/ui/CustomNodes/ScopeCardNode.tsx [hunk 3]
```diff
@@ -72,6 +74,8 @@ const ScopeCardNode = ({ data, targetPosition = Position.Top, sourcePosition = P
   const nodesMetaData = useNodesMetadata();
   const repetitionName = getRepetitionName(parentRunIndex, scopeId, nodesMetaData, operationsInfo);
   const rootRef = useRef<HTMLDivElement | null>(null);
+  const isTrigger = useMemo(() => metadata?.graphId === 'root' && metadata?.isRoot, [metadata]);
+  const panelTabName = 'SETTINGS';
```
Comment (hartra344) [inline libs/designer/src/lib/ui/CustomNodes/ScopeCardNode.tsx:78]: actually move this to a constants file somewhere, this is a true constant, not a normal use variable. This is no better than just using the string directly

File: libs/designer/src/lib/ui/CustomNodes/ScopeCardNode.tsx [hunk 4]
```diff
@@ -193,14 +197,30 @@ const ScopeCardNode = ({ data, targetPosition = Position.Top, sourcePosition = P
     WorkflowService().resubmitWorkflow?.(runInstance?.name ?? '', [id]);
   }, [runInstance, id]);
 
+  const runAfterClick = useCallback(() => {
+    dispatch(changePanelNode(scopeId));
+    dispatch(selectPanelTab(panelTabName));
+  }, [dispatch, scopeId]);
+
+  const shouldDisplayRunAfter = (operation: LogicAppsV2.ActionDefinition): boolean => {
+    if (!operation || !operation.runAfter) {
```
Comment (hartra344) [inline libs/designer/src/lib/ui/CustomNodes/ScopeCardNode.tsx:206]: move this logic somewhere central

File: libs/designer/src/lib/ui/menuItems/runAfterMenuItem.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+export const RunAfterMenuItem = (props: RunAfterMenuItemProps) => {
+  const { onClick, isTrigger = false, isScope = false } = props;
+
+  const intl = useIntl();
+
+  const runAfterMessage = intl.formatMessage({
+    defaultMessage: 'Run After',
+    id: 'rioocq',
+    description: 'Run After text',
```
Comment (hartra344) [inline libs/designer/src/lib/ui/menuItems/runAfterMenuItem.tsx:24]: please provide a better description. The description is used to provide better contextual localizations. This description provides no more information than the string itself.

File: libs/designer/src/lib/ui/menuItems/runAfterMenuItem.tsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
+import { MenuItem } from '@fluentui/react-components';
+import { bundleIcon, Copy24Filled, Copy24Regular } from '@fluentui/react-icons';
+import { LogEntryLevel, LoggerService } from '@microsoft/logic-apps-shared';
+import { useCallback } from 'react';
+import { useIntl } from 'react-intl';
+
+const RunAfterIcon = bundleIcon(Copy24Filled, Copy24Regular);
```
Comment (hartra344) [inline libs/designer/src/lib/ui/menuItems/runAfterMenuItem.tsx:7]: THis is not a good choice for ICON for this, can we please pick something better. The "Dataflows" icon would work well here I think

File: libs/designer/src/lib/ui/menuItems/runAfterMenuItem.tsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  const titleText = runAfterMessage;
+
+  const onRunAfterClick = useCallback<RunAfterMenuItemProps['onClick']>(
+    (e) => {
+      onClick(e);
+      LoggerService().log({
+        area: 'RunAfterMenuItem:onRunAfterClick',
+        args: [isScope ? 'scope' : isTrigger ? 'trigger' : 'action'],
+        level: LogEntryLevel.Verbose,
+        message: 'Clicked Run After.',
```
Comment (hartra344) [inline libs/designer/src/lib/ui/menuItems/runAfterMenuItem.tsx:36]:
```suggestion
        message: 'Clicked Run After In Context Menu.',
```

File: libs/designer/src/lib/ui/CustomNodes/ScopeCardNode.tsx [hunk 8]
```diff
@@ -193,14 +197,30 @@ const ScopeCardNode = ({ data, targetPosition = Position.Top, sourcePosition = P
...
+      return false;
+    }
+    return Object.keys(operation.runAfter).length > 0;
+  };
+
+  const operationFromWorkflow = getRecordEntry(rootState.workflow.operations, scopeId) as LogicAppsV2.OperationDefinition;
+  const runAfter = isTrigger ? false : shouldDisplayRunAfter(operationFromWorkflow);
```
Comment (hartra344) [inline libs/designer/src/lib/ui/CustomNodes/ScopeCardNode.tsx:213]: just move the trigger check into the shouldDisplayRunAfter, why keep it seperate?

**Review by hartra344 — CHANGES_REQUESTED** [review]
Also please add an E2E test or 2, you can run mocked e2e tests only by running the below command make sure you haven't generated an ARM token

`
pnpm run test:e2e --grep @mock
`

File: libs/designer/src/lib/ui/menuItems/runAfterMenuItem.tsx (hunk 7 above)
Comment (spanvalkar) [reply libs/designer/src/lib/ui/menuItems/runAfterMenuItem.tsx]: Fixed.

File: libs/designer/src/lib/ui/menuItems/runAfterMenuItem.tsx (hunk 5 above)
Comment (spanvalkar) [reply libs/designer/src/lib/ui/menuItems/runAfterMenuItem.tsx]: Fixed.

File: libs/designer/src/lib/ui/CustomNodes/ScopeCardNode.tsx (hunk 4 above)
Comment (spanvalkar) [reply libs/designer/src/lib/ui/CustomNodes/ScopeCardNode.tsx]: Fixed.

File: libs/designer/src/lib/ui/CustomNodes/ScopeCardNode.tsx (hunk 3 above)
Comment (spanvalkar) [reply libs/designer/src/lib/ui/CustomNodes/ScopeCardNode.tsx]: Fixed.

File: libs/designer/src/lib/ui/CustomNodes/OperationCardNode.tsx (hunk 2 above)
Comment (spanvalkar) [reply libs/designer/src/lib/ui/CustomNodes/OperationCardNode.tsx]: Fixed.

File: libs/designer/src/lib/ui/CustomNodes/OperationCardNode.tsx (hunk 1 above)
Comment (spanvalkar) [reply libs/designer/src/lib/ui/CustomNodes/OperationCardNode.tsx]: Fixed.

File: libs/designer/src/lib/ui/menuItems/runAfterMenuItem.tsx (hunk 6 above)
Comment (spanvalkar) [reply libs/designer/src/lib/ui/menuItems/runAfterMenuItem.tsx:7]: I cannot seem to find the 'DataFlows' icon in the fluent import files in LAUX. Does it have a specific name I should look for?

Alternatively, is there another icon you'd recommend I use instead?

**Review by hartra344 — APPROVED** [review]

**Pull request merged**